routing_common/bicycle_model.cpp

```diff
@@ -26,33 +26,33 @@ using namespace routing;
-    {HighwayType::HighwayTertiary, InOutCitySpeedKMpH(SpeedKMpH(15.0, 18.0), SpeedKMpH(20.0, 18.0))},
-    {HighwayType::HighwayTertiaryLink, InOutCitySpeedKMpH(SpeedKMpH(15.0, 18.0), SpeedKMpH(20.0, 18.0))},
-    {HighwayType::HighwayUnclassified, InOutCitySpeedKMpH(SpeedKMpH(12.0, 18.0))},
-
-    // https://github.com/organicmaps/organicmaps/issues/3881
-    // Set equal speeds here to avoid useless detours via service roads (Batumi_AvoidServiceDetour test).
-    {HighwayType::HighwayService, InOutCitySpeedKMpH(SpeedKMpH(10.0, 14.0))},
-    {HighwayType::HighwayResidential, InOutCitySpeedKMpH(SpeedKMpH(10.0, 14.0))},
-
-    {HighwayType::HighwayRoad, InOutCitySpeedKMpH(SpeedKMpH(10.0, 12.0))},
-    {HighwayType::HighwayTrack, InOutCitySpeedKMpH(SpeedKMpH(8.0, 12.0))},
-    {HighwayType::HighwayPath, InOutCitySpeedKMpH(SpeedKMpH(6.0, 12.0))},
-    {HighwayType::HighwayBridleway, InOutCitySpeedKMpH(SpeedKMpH(4.0, 12.0))},
-    {HighwayType::HighwayCycleway, InOutCitySpeedKMpH(SpeedKMpH(30.0, 20.0))},
-    {HighwayType::HighwayLivingStreet, InOutCitySpeedKMpH(SpeedKMpH(7.0, 8.0))},
+    // Note that roads with hwtag=yesbicycle get high speed of 0.9 * Cycleway.
+    /// @see Russia_UseTrunk test for Trunk weights.
+    {HighwayType::HighwayTrunk, InOutCitySpeedKMpH(SpeedKMpH(7.0, 17.0), SpeedKMpH(9.0, 19.0))},
+    {HighwayType::HighwayTrunkLink, InOutCitySpeedKMpH(SpeedKMpH(7.0, 17.0), SpeedKMpH(9.0, 19.0))},
+    {HighwayType::HighwayPrimary, InOutCitySpeedKMpH(SpeedKMpH(10.0, 17.0), SpeedKMpH(12.0, 19.0))},
+    {HighwayType::HighwayPrimaryLink, InOutCitySpeedKMpH(SpeedKMpH(10.0, 17.0), SpeedKMpH(12.0, 19.0))},
+    {HighwayType::HighwaySecondary, InOutCitySpeedKMpH(SpeedKMpH(13.0, 17.0), SpeedKMpH(15.0, 19.0))},
+    {HighwayType::HighwaySecondaryLink, InOutCitySpeedKMpH(SpeedKMpH(13.0, 17.0), SpeedKMpH(15.0, 19.0))},
+    {HighwayType::HighwayTertiary, InOutCitySpeedKMpH(SpeedKMpH(14.0, 17.0), SpeedKMpH(17.0, 19.0))},
+    {HighwayType::HighwayTertiaryLink, InOutCitySpeedKMpH(SpeedKMpH(14.0, 17.0), SpeedKMpH(17.0, 19.0))},
+    {HighwayType::HighwayUnclassified, InOutCitySpeedKMpH(SpeedKMpH(13.0, 17.0), SpeedKMpH(15.0, 19.0))},
+    {HighwayType::HighwayResidential, InOutCitySpeedKMpH(SpeedKMpH(12.0, 15.0), SpeedKMpH(14.0, 17.0))},
+    {HighwayType::HighwayService, InOutCitySpeedKMpH(SpeedKMpH(10.0, 12.0), SpeedKMpH(14.0, 17.0))},
+    {HighwayType::HighwayRoad, InOutCitySpeedKMpH(SpeedKMpH(10.0, 12.0), SpeedKMpH(14.0, 17.0))},
+
+    {HighwayType::HighwayTrack, InOutCitySpeedKMpH(SpeedKMpH(8.0, 12.0), SpeedKMpH(10.0, 14.0))},
+    {HighwayType::HighwayPath, InOutCitySpeedKMpH(SpeedKMpH(6.0, 10.0), SpeedKMpH(7.0, 12.0))},
+    {HighwayType::HighwayBridleway, InOutCitySpeedKMpH(SpeedKMpH(4.0, 10.0), SpeedKMpH(5.0, 12.0))},
+
+    {HighwayType::HighwayCycleway, InOutCitySpeedKMpH(SpeedKMpH(21.0, 18.0), SpeedKMpH(23.0, 20.0))},
+    {HighwayType::HighwayLivingStreet, InOutCitySpeedKMpH(SpeedKMpH(8.0, 8.0))},
```

Comment:
why living street is so heavily penalised? more than `highway=primary` ?

maybe real speed will be fairly low (but still greater than 8, I bet) but I would definitely take slower driving on `highway=living_street` over `highway=primary`

some will be extremely busy with pedestrians, but most are great for cycling (and ones most crowded will often be `bicycle=no` anyway)

```suggestion
    {HighwayType::HighwayLivingStreet, InOutCitySpeedKMpH(SpeedKMpH(13.0, 10.0))},
```

this matches `highway=secondary` in desirability and path/bridleway in speed

Reply:
Sounds reasonable, I'll try to adjust and see if the tests are good. Would be great also to add specific tests for living_street preference/avoidance cases.

IDK the reason it was penalized before, I can only guess it stems from experience of living_streets in ex-USSR countries used predominantly for inside-the-block small and winding access roads.

@vng might know more.

Reply:
meandering roads should be solved by itself due to increased length...


bad: https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=50.053595%2C19.948463%3B50.050640%2C19.950151#map=17/50.052076/19.949938&layers=N
good: https://www.openstreetmap.org/directions?engine=fossgis_osrm_bike&route=50.053595%2C19.948463%3B50.050640%2C19.950151#map=17/50.052062/19.949316&layers=N
(going together with trams, busses and heavy traffic is clearly inferior)

https://www.openstreetmap.org/directions?engine=fossgis_osrm_bike&route=50.006085%2C19.962158%3B50.006664%2C19.957639#map=18/50.006384/19.959900&layers=N
currently OM wants to do detour via south (warning: not checked this area in person, looks like a good test based on mapped data)

https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=50.020758%2C20.049156%3B50.022031%2C20.048289 (OM wants illegally drive on sidewalk, GraphHopper wants to go on route that is of lower quality but balanced by being longer :) )

looks like good test case, never was there:

https://www.openstreetmap.org/directions?engine=graphhopper_bicycle&route=50.096027%2C19.904330%3B50.099875%2C19.889867 

should be

https://www.openstreetmap.org/directions?engine=fossgis_osrm_bike&route=50.096008%2C19.904225%3B50.099847%2C19.889835

Reply:
Forgot to add that OM with latest changes passes 3 cases you've suggested.

Reply:
> https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=50.020758%2C20.049156%3B50.022031%2C20.048289 (OM wants illegally drive on sidewalk, GraphHopper wants to go on route that is of lower quality but balanced by being longer :) )

OM goes via a sidewalk still because it has `bicycle=yes`. Actually it might be a good test case for #9593

Reply:
> OM goes via a sidewalk still because it has bicycle=yes

Interesting, I opened https://www.openstreetmap.org/changeset/151976870

Reply:
> Forgot to add that OM with latest changes passes 3 cases you've suggested.

congratulations!

Reply:
> OM goes via a sidewalk still because it has bicycle=yes

contacted the mapper in https://www.openstreetmap.org/changeset/151976870 - they commented that it was a mistake and removed it